packages/node/src/typescript.ts

```diff
@@ -502,8 +522,13 @@ export function fixConfig(
   }
 
   // If not specified, the default Node.js module is CommonJS.
+  // However, if package.json has "type": "module" or the file is .mts, use ESNext instead.
   if (config.compilerOptions.module === undefined) {
-    config.compilerOptions.module = 'CommonJS';
+    if ((pkg?.type === 'module' && moduleTypeExtension !== 'cts') || moduleTypeExtension === 'mts') {
+      config.compilerOptions.module = 'ESNext';
+    } else {
+      config.compilerOptions.module = 'CommonJS';
```

Comment:
I wonder, can we solve this problem by setting `nodenext` instead of `esnext`?

Since `nodenext` is supposed to consider the package.json type and file extension.

Reply:
This works, the validation around this is mostly sitting in the express/hono builder tests, but will move some over once the dust settle.

It was failing on some tests but it was because for some reason `nodenext` seems to also turn on `strict` mode. I can't find any mention of that in documentation, but I also think it's safe to explicitly set `strict: false` here anyway.

Reply:
Thats odd. I wouldn’t expect strict mode to change. Maybe it’s the “use strict” for modules as opposed to TS strict:true.